diff -up gst-libav-1.24.5/ext/libav/gstavcfg.c.1~ gst-libav-1.24.5/ext/libav/gstavcfg.c
diff -up gst-libav-1.24.5/ext/libav/gstavprotocol.c.1~ gst-libav-1.24.5/ext/libav/gstavprotocol.c
--- gst-libav-1.24.5/ext/libav/gstavprotocol.c.1~	2024-06-20 13:54:15.000000000 +0200
+++ gst-libav-1.24.5/ext/libav/gstavprotocol.c	2024-06-21 16:47:18.899715185 +0200
@@ -102,7 +102,11 @@ gst_ffmpegdata_read (void *priv_data, un
 }
 
 static int
-gst_ffmpegdata_write (void *priv_data, uint8_t * buf, int size)
+gst_ffmpegdata_write (void *priv_data,
+#if LIBAVCODEC_VERSION_MAJOR > 60
+ const 
+#endif
+  uint8_t * buf, int size)
 {
   GstProtocolInfo *info;
   GstBuffer *outbuf;
diff -up gst-libav-1.24.5/ext/libav/gstavviddec.c.1~ gst-libav-1.24.5/ext/libav/gstavviddec.c
diff -up gst-libav-1.24.5/ext/libav/gstavvidenc.c.1~ gst-libav-1.24.5/ext/libav/gstavvidenc.c
