diff -uraN gst-libav-1.24.6/ext/libav/gstavprotocol.c omv-gst-libav-1.24.6/ext/libav/gstavprotocol.c
--- gst-libav-1.24.6/ext/libav/gstavprotocol.c	2024-07-29 17:41:37.000000000 +0200
+++ omv-gst-libav-1.24.6/ext/libav/gstavprotocol.c	2024-08-01 20:19:17.126814693 +0200
@@ -102,11 +102,11 @@
 }
 
 static int
-#if LIBAVUTIL_VERSION_MAJOR >= 59
-gst_ffmpegdata_write (void *priv_data, const uint8_t * buf, int size)
-#else
-gst_ffmpegdata_write (void *priv_data, uint8_t * buf, int size)
+gst_ffmpegdata_write (void *priv_data,
+#if LIBAVCODEC_VERSION_MAJOR > 60
+ const
 #endif
+  uint8_t * buf, int size)
 {
   GstProtocolInfo *info;
   GstBuffer *outbuf;
